app/models/subnet.rb

```diff
@@ -216,6 +220,12 @@ def unused_ip(mac = nil, excluded_ips = [])
       end
       logger.debug("Not suggesting IP Address for #{self} as no free IP found in our DB")
       return
+    elsif self.ipam == IPAM_MODES[:eui64] && self.class.supports_ipam_mode?(:eui64)
```

Comment:
The whole `unused_ip` method should be implemented on `Subnet::Ipv4` and `Subnet::Ipv6`. What these `if` expressions are doing is basically class checking. 

`dhcp` and `db` should be in `unused_ip` `Subnet::Ipv4` and `eui64` should be in `unused_ip` in `Subnet::Ipv6`. There would be 3 lines of duplicated code at the beginning, lines 197 to 200 of this patch but that should be alright